stack_overflow/src/model/User.js

```diff
@@ -0,0 +1,110 @@
+import { EventEmitter } from "events";
+
+class User extends EventEmitter{
+
+    constructor() {
+        super();
+        this.state = {
+            users: [{
+                id: 1,
+                username: "user1",
+                password: "pass1",
+                email: "email1",
+                score: 0,
+                isAdmin: true,
+                isBanned: false,
+            }, {
+                id: 2,
+                username: "user2",
+                password: "pass2",
+                email: "email2",
+                score: 0,
+                isAdmin: false,
+                isBanned: false,
+            }],
+            newUser: {
+                id: 3,
+                username: "",
+                password: "",
+                email: "",
+                score: 0,
+                isAdmin: false,
+                isBanned: false,
+            },
+
+            loggedUser: {
+                id: 1,
+                username: "user1",
+                password: "pass1",
+                email: "email1",
+                score: 0,
+                isAdmin: true,
+                isBanned: false,
+            },
+
+            currentIndex: 3
+        };
+    }
+
+    addUser(username, password, email, score, isAdmin, isBanned) {
+        this.state = {
+            ...this.state,
+            users: this.state.users.concat([{
+                id: this.state.currentIndex,
+                username: username,
+                password: password,
+                email: email,
+                score: score,
+                isAdmin: isAdmin,
+                isBanned: isBanned
+            }])
+        };
+        this.state.currentIndex = this.state.currentIndex + 1;
```

Comment:
You are mutating the state object. In this case, it should not do any harm, but you should have done this at line 50 where you created a clone of the state.